lib/netstd/Thrift/Transport/Client/THttpTransport.cs

```diff
@@ -172,41 +166,41 @@ public override async Task FlushAsync(CancellationToken cancellationToken)
         {
             try
             {
-                try
+                _outputStream.Seek(0, SeekOrigin.Begin);
+
+                using (var contentStream = new StreamContent(_outputStream))
                 {
-                    if (_outputStream.CanSeek)
-                    {
-                        _outputStream.Seek(0, SeekOrigin.Begin);
-                    }
+                    contentStream.Headers.ContentType = ContentType ?? new MediaTypeHeaderValue(@"application/x-thrift");
 
-                    using (var outStream = new StreamContent(_outputStream))
+                    try
                     {
-                        var msg = await _httpClient.PostAsync(_uri, outStream, cancellationToken);
-
-                        msg.EnsureSuccessStatusCode();
-
-                        if (_inputStream != null)
-                        {
-                            _inputStream.Dispose();
-                            _inputStream = null;
-                        }
+                        var response = (await _httpClient.PostAsync(_uri, contentStream, cancellationToken)).EnsureSuccessStatusCode();
 
-                        _inputStream = await msg.Content.ReadAsStreamAsync();
+                        _inputStream?.Dispose();
+                        _inputStream = await response.Content.ReadAsStreamAsync();                        
                         if (_inputStream.CanSeek)
                         {
                             _inputStream.Seek(0, SeekOrigin.Begin);
                         }
                     }
+                    catch (HttpRequestException rex)
+                    {
+                        throw new TTransportException(TTransportException.ExceptionType.Unknown, $"Failed to connect: {rex.Message}");
+                    }
+                    catch (Exception ex)
+                    {
+                        throw new TTransportException(TTransportException.ExceptionType.Unknown, ex.Message);
+                    }
                 }
-                catch (IOException iox)
-                {
-                    throw new TTransportException(TTransportException.ExceptionType.Unknown, iox.ToString());
-                }
-                catch (HttpRequestException wx)
-                {
-                    throw new TTransportException(TTransportException.ExceptionType.Unknown,
-                        "Couldn't connect to server: " + wx);
-                }
+            }
+            catch (IOException iox)
+            {
+                throw new TTransportException(TTransportException.ExceptionType.Unknown, iox.ToString());
+            }
+            catch (HttpRequestException wx)
+            {
+                throw new TTransportException(TTransportException.ExceptionType.Unknown,
+                    "Couldn't connect to server: " + wx);
             }
```

Comment:
I have a general problem with nested exception handlers. There are exceptions to the rule (pun intended) but in most cases they are just badly designed and clutter the code towards unmaintainability. In the particular case we have here we have three different tpyes of exceptions which are handled in a strange manner. Is there a chance we can streamline that a bit and consolidate all catches into one level? Or do I overlook sth important?

Reply:
Agreed. Looking through the code and I don't see any benefit to the nested try-catch statements. The duplicate catch for HttpRequestException  wouldn't ever be hit as far as I could tell. I moved them all into the same level and removed the other nested try-finally since it too could be put in line. Let me know if I went to far.